### DISCUSSION TITLE
Speed Test inside and Outside of the Container

### DISCUSSION THREAD

**Discussion by Stampede** [haugene/docker-transmission-openvpn#2557] [Ideas]
I was getting pretty slow transfer speeds yesterday and started tinkering around.
I wound up entering a bash shell in the docker container and did `# apt install speedtest-cli` I also installed `speedtest-cli` on the host machine. It's basically a CLI version of the speedtest.net website.

So now I can compare network transfer speeds inside and outside of the container to see how much the VPN is slowing me down. Then I shut down the container to change the docker-compose.yaml file and change the VPN server the container connects to, which did improve my transfer speeds....for now.
Private internet access provides a script for testing their server latencies: https://github.com/pia-foss/manual-connections/blob/master/get_region.sh

I thought about trying to write a script that occasionally ran the PIA server latency script, and then switched to a different PIA server if the currently connected one  was too slow, but when I restarted the docker container, I had to reinstall speedtest-cli

Also, I'm not sure the best way to change servers...if it's to shut down the docker container, update the compose.yaml, then restart, or if there's a more elegent way to do it from within the container.

In any case, such a script wouldn't be much use without a command line speed test inside the container. Maybe it's worth shipping future containers with something like speedtest-cli?

Open to hear any thoughts. I know just enough about VPNs to be dangerous, so maybe this isn't a very good idea.

**Comment by pkishino** [comment]
I think this was mentioned in the wiki..anyway,
no, no plans to include the speedtest-cli to the image as that is not the purpose of this image..
testing speeds based on vpn includes a number of variables etc..the better place to do this would be to have a script on the provider level and then perform such a check etc based on provider recommendations. Some providers in our repo have scripts, others not. Some provide random servers, location profiles etc etc..
Yes, ideally, the vpn-provider-repo would have scripts for each provider supported, when ran it would do a speediest based on current location or whatever and then select the best server for you...feel free to help out on that front ;)
